service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -70,6 +80,53 @@ private void prePopulateHearingDateSelection(List<Element<HearingBooking>> heari
         hearingDatesDynamic.setValue(listElement);
     }
 
+    public String createRespondentAssigneeDropdownKey(CaseDetails caseDetails) {
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (isNotEmpty(caseData.getRespondents1())) {
+            for (int i = 0; i < caseData.getRespondents1().size(); i++) {
+                RespondentParty respondentParty = caseData.getRespondents1().get(i).getValue().getParty();
+
+                String key = String.format("Respondent %d - %s", i + 1, getRespondentsFullName(respondentParty));
+                stringBuilder.append(key).append("\n\n");
+            }
+        }
+
+        return stringBuilder.toString().stripTrailing();
+    }
+
+    public String createOtherPartiesAssigneeDropdownKey(CaseDetails caseDetails) {
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (isNotEmpty(caseData.getOthers())) {
+            for (int i = 0; i < caseData.getOthers().getAllOthers().size(); i++) {
+                Other other = caseData.getOthers().getAllOthers().get(i);
+                String key;
+
+                if (i == 0) {
+                    key = String.format("Person %d - %s", i + 1, defaultIfNull(other.getName(), ""));
+                } else {
+                    key = String.format("Other Person %d - %s", i + 1, defaultIfNull(other.getName(), ""));
+                }
+
+                stringBuilder.append(key).append("\n\n");
+            }
+        }
+
+        return stringBuilder.toString().stripTrailing();
+    }
+
+    private String getRespondentsFullName(RespondentParty respondentParty) {
+        String firstName = defaultIfNull(respondentParty.getFirstName(), "");
+        String lastName = defaultIfNull(respondentParty.getLastName(), "");
+
+        return firstName + " " + lastName;
```

Comment:
Can we use StringBuilder to concatenate string here?

Reply:
I can use string.format. I think stringBuilder is a bit of overkill for this.